hydra-cluster/bench/Bench/EndToEnd.hs

```diff
@@ -95,57 +96,116 @@ bench startingNodeId timeoutSeconds workDir dataset@Dataset{clientDatasets, titl
+              scenario hydraTracer node workDir dataset parties leader followers
+
+benchDemo ::
+  NetworkId ->
+  SocketPath ->
+  NominalDiffTime ->
+  VerificationKey PaymentKey ->
+  [SigningKey HydraKey] ->
+  FilePath ->
+  Dataset ->
+  IO Summary
+benchDemo networkId nodeSocket timeoutSeconds faucetVk hydraKeys workDir dataset@Dataset{clientDatasets, fundingTransaction} = do
+  putStrLn $ "Test logs available in: " <> (workDir </> "test.log")
+  withFile (workDir </> "test.log") ReadWriteMode $ \hdl ->
+    withTracerOutputTo hdl "Test" $ \tracer ->
+      failAfter timeoutSeconds $ do
+        putTextLn "Starting benchmark"
+        let cardanoTracer = contramap FromCardanoNode tracer
+        findRunningCardanoNode' cardanoTracer networkId nodeSocket >>= \case
+          Nothing ->
+            error ("Not found running node at socket: " <> show nodeSocket <> ", and network: " <> show networkId)
+          Just node -> do
+            let clientSks = clientKeys <$> clientDatasets
+            (`finally` returnFaucetFunds tracer node clientSks) $ do
+              putTextLn "Seeding network"
+              fundClients networkId nodeSocket fundingTransaction
+              forM_ clientSks (fuelWith100Ada (contramap FromFaucet tracer) node)
+              putStrLn $ "Connecting to hydra cluster in " <> workDir
+              let hydraTracer = contramap FromHydraNode tracer
+              let parties = Set.fromList (deriveParty <$> hydraKeys)
+              withConnectionToNode hydraTracer 1 $ \leader ->
+                withConnectionToNode hydraTracer 2 $ \node2 ->
+                  withConnectionToNode hydraTracer 3 $ \node3 -> do
+                    let followers = [node2, node3]
+                    scenario hydraTracer node workDir dataset parties leader followers
+ where
+  returnFaucetFunds tracer node cKeys = do
+    putTextLn "Returning funds to faucet"
+    let faucetTracer = contramap FromFaucet tracer
+    let toSenders (ClientKeys sk esk) = [(getVerificationKey sk, sk), (getVerificationKey esk, esk)]
+    let senders = concatMap @[] toSenders cKeys
+    mapM_
+      ( \sender -> do
+          returnAmount <- returnFundsToFaucet' faucetTracer node faucetVk sender
+          traceWith faucetTracer $ ReturnedFunds{actor = show sender, returnAmount}
+      )
+      senders
+
+scenario ::
+  Tracer IO HydraNodeLog ->
+  RunningNode ->
+  FilePath ->
+  Dataset ->
+  Set Party ->
+  HydraClient ->
+  [HydraClient] ->
+  IO Summary
+scenario hydraTracer node workDir dataset@Dataset{clientDatasets, title, description} parties leader followers = do
+  let clusterSize = fromIntegral $ length clientDatasets
+  let clients = leader : followers
+  waitForNodesConnected hydraTracer 20 clients
+
+  putTextLn "Initializing Head"
+  send leader $ input "Init" []
+  headId <-
+    waitForAllMatch (fromIntegral $ 10 * clusterSize) clients $
+      headIsInitializingWith parties
```

Comment:
This is the only location where you actually use the `--hydra-sk` options. Instead of going at lengths to provide them, we could just not care too much about what head we see initializing. For example we could just check it's a head with the expected number of participants (`length (leader:followers)`)

Should: not need `--hydra-sk` just for this reason.